tests/test_extensions.py

```diff
@@ -0,0 +1,56 @@
+# This file is part of astro_metadata_translator.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the LICENSE file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# Use of this source code is governed by a 3-clause BSD-style
+# license that can be found in the LICENSE file.
+
+import pickle
+import unittest
+from astro_metadata_translator import ObservationInfo, makeObservationInfo, PropertyDefinition
+
+"""Test that extensions to the core set of properties works"""
+
+
+class ExtensionsTestCase(unittest.TestCase):
+    def setUp(self):
+        self.extensions = dict(
+            myInfo=PropertyDefinition("My special information", "str", str),
+        )
+
+    def test_basic(self):
+        """Test construction of extended ObservationInfo"""
+        string = "the usual"
+        obsinfo = makeObservationInfo(extensions=self.extensions, ext_myInfo=string)
+
+        # Behaves like the original
+        self.assertIsInstance(obsinfo, ObservationInfo)
+        self.assertEqual(obsinfo.ext_myInfo, string)
+
+        with self.assertRaises(AttributeError):
+            # Variable is read-only
+            obsinfo.ext_myInfo = "something completely different"
+
+        with self.assertRaises(KeyError):
+            # Can't specify extension value without declaring extensions
+            obsinfo.makeObservationInfo(ext_myInfo="foobar")
+
+        with self.assertRaises(TypeError):
+            # Type checking is applied, like in the original
+            obsinfo.makeObservationInfo(extensions=self.extensions, ext_myInfo=12345)
+
+    def test_pickle(self):
```

Comment:
We will need to show that JSON round trips and you can serialize an ObservationInfo to JSON and then read the JSON back in and get an equivalent ObservationInfo.